data/src/main/java/com/linkedin/data/DataComplexTable.java

```diff
@@ -29,11 +29,36 @@
 
   public DataComplex get(DataComplex index)
   {
-    return _map.get(index.dataComplexHashCode());
+    return _map.get(new DataComplexKey(index));
   }
 
   public void put(DataComplex src, DataComplex clone)
   {
-    _map.put(src.dataComplexHashCode(), clone);
+    _map.put(new DataComplexKey(src), clone);
   }
+
+  private static class DataComplexKey
+  {
+    private final DataComplex _dataObject;
+    private final int _hashCode;
+
+    DataComplexKey(DataComplex dataObject)
+    {
+      _hashCode = dataObject.dataComplexHashCode();
+      _dataObject = dataObject;
+    }
+
+    @Override
+    public int hashCode()
+    {
+      return _hashCode;
+    }
+
+    @Override
+    public boolean equals(Object other)
+    {
+      return _dataObject == ((DataComplexKey) other)._dataObject;
```

Comment:
Should there be some usual check for the object?
Something like the equals override in this link
https://www.baeldung.com/java-hashcode

Reply:
Not needed as the other object will always be DataComplex. This Key type will only be used within DataComplexTable. 
Added a comment to explain this.